backend/src/test/java/put/eunice/cms/resource/ResourceControllerTest.java

```diff
@@ -0,0 +1,325 @@
+            }
+        }
+
+        @Nested
+        class CreateResourceTestClass {
+
+            @Test
+            void create_Resource_GuestUser_unauthorized() throws Exception {
+                ResourceDtoFormCreate dto =
+                        new ResourceDtoFormCreate("name", "desc", userId, createMockMultipartFile(), null);
+                performAsGuest();
+                performPostFile(dto).andExpect(status().isUnauthorized());
+            }
+
+            @Test
+            void create_Resource_User_Success() throws Exception {
+                performAs(Role.USER, userId);
+                ResourceDtoFormCreate dto =
+                        new ResourceDtoFormCreate("name", "desc", userId, createMockMultipartFile(), null);
+                performPostFile(dto).andExpect(status().isOk());
+            }
+
+            @Test
+            void create_Resource_Moderator_Success() throws Exception {
+                performAs(Role.MODERATOR, userId);
+                ResourceDtoFormCreate dto =
+                        new ResourceDtoFormCreate("name", "desc", userId, createMockMultipartFile(), null);
+                performPostFile(dto).andExpect(status().isOk());
+            }
+
+            @Test
+            void create_Resource_Admin_Success() throws Exception {
+                performAs(Role.ADMIN, userId);
+                ResourceDtoFormCreate dto =
+                        new ResourceDtoFormCreate("name", "desc", userId, createMockMultipartFile(), null);
+                performPostFile(dto).andExpect(status().isOk());
+            }
+
+            @Test
+            void create_Link_GuestUser_unauthorized() throws Exception {
+                ResourceDtoFormCreate dto = new ResourceDtoFormCreate("name", "desc", userId, null, "URL");
+                performAsGuest();
+                performPostFile(dto).andExpect(status().isUnauthorized());
+            }
+
+            @Test
+            void create_Link_User_Success() throws Exception {
+                performAs(Role.USER, userId);
+                ResourceDtoFormCreate dto = new ResourceDtoFormCreate("name", "desc", userId, null, "URL");
+                performPostFile(dto).andExpect(status().isOk());
+            }
+
+            @Test
+            void create_Link_Moderator_Success() throws Exception {
+                performAs(Role.MODERATOR, userId);
+                ResourceDtoFormCreate dto = new ResourceDtoFormCreate("name", "desc", userId, null, "URL");
+                performPostFile(dto).andExpect(status().isOk());
+            }
+
+            @Test
+            void create_Link_Admin_Success() throws Exception {
+                performAs(Role.ADMIN, userId);
+                ResourceDtoFormCreate dto = new ResourceDtoFormCreate("name", "desc", userId, null, "URL");
+                performPostFile(dto).andExpect(status().isOk());
+            }
+        }
+
+        @Nested
+        class UpdateResourceTestClass {
```

Comment:
W update jedyne czego mi brakuje to żeby sprawdzić, czy jak robie update czegoś co jest już linkiem to czy po update nadal jest linkiem i tak samo z typem = plik, bo jak testowałem ręcznie to jeśli robiłem update typu plik nie wrzucając nowego pliku (tylko nowe description) to zmieniało typ na link i jako link dawało string "null". Teraz już powinno działać, ale taki test by się przydał dla pewności